x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/SecuritySearchOperationListener.java

```diff
@@ -40,12 +41,12 @@
 public final class SecuritySearchOperationListener implements SearchOperationListener {
 
     private final SecurityContext securityContext;
-    private final XPackLicenseState licenseState;
     private final AuditTrailService auditTrailService;
+    private final Settings settings;
```

Comment:
This is similar to the `SecurityIndexReaderWrapper` case, we don't need to look at the settings.

If `xpack.security.enabled` is `false` the `SecuritySearchOperationListener` ought not be created and the `Security` plugin's `onIndexModule` should be a no-op.